### ISSUE TITLE
Add section about object store data source

### ISSUE THREAD

**Issue by jhonabreul** [QuantConnect/Documentation#1500]
#### Expected Behavior

Document how to use the object store as a data source using:

- Text data ([demo](https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_fcf3336724e95d00eda1fc141e108b55.html))
- Zipped data ([demo](https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_1a3d4f56ab79816e69aadca68008a7d9.html))
- Multi-file data ([demo](https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_bd883b6368fca7b861dba6067f06a143.html))

The linked demo algorithms showcase how to source data stored in the object store and how to use it both in algorithms and research environments, by using the new subscription transport medium `SubscriptionTransportMedium.ObjectStore`. They show case two use cases, streaming the data for the algorithms and in history requests.

See new regression algorithms in https://github.com/QuantConnect/Lean/pull/7493

#### Actual Behavior

New feature, not documented.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

**Comment by AlexCatarino** [comment]
On [Streaming Data > Key Concepts](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/streaming-data/key-concepts):
- In the Introduction, say that we can stream data over the internet or from the Object Store.
- Add a H3 after the Introduction to discuss the Object Store. We need to say that members can upload the data using Download and save to the ObjectStore using Save or use CLI or Object Store explorer. We can use csv-zipped files.
- In Data Format, rephrase the following
> Each request has a one-second overhead, so you should package your custom data to minimize requests. Bundle dates together where possible to speed up execution.

because making requests over the internet is an option now. We can recommend using the Object Store for speed.

On [Custom Securities > CSV Format Example](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/streaming-data/custom-securities/csv-format-example), [Custom Securities > JSON Format Example](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/streaming-data/custom-securities/json-format-example), [Custom Universes > CSV Format Example](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/streaming-data/custom-universes/csv-format-example), and [Custom Universes > JSON Format Example](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/streaming-data/custom-universes/json-format-example), add H3 after "Data Format" (e.g.: "Saving Data") to inform that the data can be saved to the Object Store.

All Define Custom Types should have:
```python
    def GetSource(self, config: SubscriptionDataConfig, date: datetime, isLive: bool) -> SubscriptionDataSource:
        if not isLive:
            return SubscriptionDataSource(CustomDataKey, 
                SubscriptionTransportMedium.ObjectStore, 
                FileFormat.Csv)
        return SubscriptionDataSource("https://www.dropbox.com/s/ae1couew5ir3z9y/daily-stock-picker-backtest.csv?dl=1",
            SubscriptionTransportMedium.RemoteFile)
```
to show the ObjectStore usage as "default" for backtest.

On [Bulk Downloads](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/bulk-downloads), add losing data to use with the ObjectStore in "Recommended Use Cases".

After [Download Files](https://www.quantconnect.com/docs/v2/writing-algorithms/importing-data/bulk-downloads#03-Download-Files), add a section teaching how to use Download+ObjectStore.Save to save a zip file to be used in streaming.

```python
CustomDataKey = "CustomData20231010/ExampleCustomData.zip"
data = self.Download("https://www.dropbox.com/s/ae1couew5ir3z9y/daily-stock-picker-backtest.csv?dl=1")
self.ObjectStore.SaveBytes(CustomDataKey,
    Compression.ZipBytes(bytes(self.CustomData, 'utf-8'), 'data'))
```

**Comment by AlexCatarino** [comment]
Closed by #1511
